bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/AutoUpdateManager.java

```diff
@@ -281,14 +289,19 @@ private boolean isAcceptedState(State newState, Item item) {
             // Look for class hierarchy
             for (Class<?> state : item.getAcceptedDataTypes()) {
                 try {
-                    if (!state.isEnum() && state.newInstance().getClass().isAssignableFrom(newState.getClass())) {
+                    if (!state.isEnum() && state.getDeclaredConstructor().newInstance().getClass()
+                            .isAssignableFrom(newState.getClass())) {
                         isAccepted = true;
                         break;
                     }
                 } catch (InstantiationException e) {
                     logger.warn("InstantiationException on {}", e.getMessage(), e); // Should never happen
                 } catch (IllegalAccessException e) {
                     logger.warn("IllegalAccessException on {}", e.getMessage(), e); // Should never happen
+                } catch (NoSuchMethodException e) {
+                    logger.warn("NoSuchMethodException on {}", e.getMessage(), e); // Should never happen
+                } catch (InvocationTargetException e) {
+                    logger.warn("InvocationTargetException on {}", e.getMessage(), e); // Should never happen
```

Comment:
I guess we can shrink this block to be a one liner:

```suggestion
                } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
                    logger.warn("Exception on {}", e.getMessage(), e); // Should never happen
```